codalab/model/tables.py

```diff
@@ -327,3 +328,58 @@
   Column('bundle_uuid', String(63), nullable=True), # What is the id of the bundle that the sender is on? 
   sqlite_autoincrement=True,
 )
+
+# Store information about workers.
+worker = Table(
+  'worker',
+  db_metadata,
+
+  Column('user_id', String(63), ForeignKey(user.c.user_id), primary_key=True, nullable=False),
+  Column('worker_id', String(127), primary_key=True, nullable=False),
+
+  Column('slots', Integer, nullable=False),  # Number of additional bundles the worker can run.
+  Column('checkin_time', DateTime, nullable=False),  # When the worker last checked in with the bundle service.
+  Column('socket_id', Integer, nullable=False),  # Socket ID worker listens for messages on.
+)
+
+# Store information about all sockets currently allocated to each worker.
+worker_socket = Table(
+  'worker_socket',
+  db_metadata,
+
+  Column('user_id', String(63), ForeignKey(user.c.user_id), nullable=False),
+  Column('worker_id', String(127), nullable=False),
+  # No foreign key constraint on the worker table so that we can create a socket
```

Comment:
Why can't we just create the worker entry before worker_socket (we can just fill in slots, checkin_time, socket_id with null-ish values)?